posts/2024-08-27-24.0.0.9-beta.adoc

```diff
@@ -0,0 +1,212 @@
+
+For example, the `ConnectionPool` component requires the following configuration:
+
+[source,xml]
+----
+<featureManager>
+   <feature>mpTelemetry-2.0</feature>
+   <feature>jdbc-4.3</feature>
+</featureManager>
+----
+
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC>
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/29332
+// Contact/Reviewer: pgunapal
+// // // // // // // //
+[#logs]
+== Send logs to OpenTelemetry
+
+The `mpTelemetry-2.0` feature can now collect Open Liberty runtime log sources (messages, traces, ffdcs) and application logs generated through the `java.util.logging` package (JUL).
+
+To enable the MicroProfile Telemetry 2.0 feature to collect all logs, add the following configuration to your `server.xml` file:
+
+[source,xml]
+----
+<features>
+   <feature>mpTelemetry-2.0</feature>
+</features>
+
+<mpTelemetry source="message, trace, ffdc"/>
+----
+
+If the `mpTelemetry` configuration element or the `source` attribute is not configured, the `message` source is set by default. In this case, only messages are collected. If the `source` attribute is specified empty (`source=""`), no logs are sent to OpenTelemetry.
+
+To collect and export runtime-level logs, enable OpenTelemetry by using the following system property or environment variable:
+
+* System property: `otel.sdk.disabled=false`
+* Environment variable: `OTEL_SDK_DISABLED=false`
+
+You can set configuration properties in any of the link:{url-prefix}/docs/latest/external-configuration.html#default[config sources that are available to MicroProfile Config].
+
+To separately configure multiple applications in a server, you can configure OpenTelemetry with application configuration. However, you cannot collect runtime-level logs this way.
+
+By default, all OpenTelemetry data is exported to link:https://opentelemetry.io/docs/languages/java/exporters/#otlp[OTLP]. You set a different exporter by specifying the following system property or environment variable:
+
+* System property: `otel.logs.exporter`
+* Environment variable: `OTEL_LOGS_EXPORTER`
+
+// DO NOT MODIFY THIS LINE. </GHA-BLOG-TOPIC>
+
+// // // // DO NOT MODIFY THIS COMMENT BLOCK <GHA-BLOG-TOPIC> // // // //
+// Blog issue: https://github.com/OpenLiberty/open-liberty/issues/29223
+// Contact/Reviewer: volosied,pnicolucci
+// // // // // // // //
+[#websocket]
+== Avoid type errors with WebSocket 2.2
+
+To avoid type errors, this beta release of the WebSocket 2.2 feature (`websocket-2.2`) limits the link:https://jakarta.ee/specifications/websocket/2.2/apidocs/server/jakarta/websocket/onmessage#maxMessageSize([MaxMessageSize]) to `Integer#MAX_VALUE` for the `OnMessage` annotation. The websocket-2.2 feature was previously released in beta in the 24.0.0.8-beta.
```

Comment:
Can you add a ending parenthesis **()** to the jakarta.ee link so that it takes you directly to the maxMessageSize section of the page? 

https://jakarta.ee/specifications/websocket/2.2/apidocs/server/jakarta/websocket/onmessage#maxMessageSize()

Reply:
done